src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1330,6 +1331,32 @@ citus_update_node(PG_FUNCTION_ARGS)
 }
 
 
+Datum
+citus_pause_node(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	int32 nodeId = PG_GETARG_INT32(0);
+	List *placementList = NIL;
+
+	WorkerNode *workerNode = FindNodeAnyClusterByNodeId(nodeId);
+	if (workerNode == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_NO_DATA_FOUND),
+						errmsg("node %u not found", nodeId)));
+	}
+
+
+	if (NodeIsPrimary(workerNode))
```

Comment:
why only primary nodes? Can't we pause traffic to read-replicas? Is there anything that prevents us?

Reply:
the original code in citus_update_node has a comment about this:
```
	 * It might be worth blocking reads to a secondary for the same reasons,
	 * though we currently only query secondaries on follower clusters
	 * where these locks will have no effect.
```